src/App/Exercises/Js/Try9/TodoItem.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react';
+import './TodoItem.css';
+import DeleteIcon from '@mui/icons-material/Delete';
+import DoneIcon from '@mui/icons-material/Done';
+import EditIcon from '@mui/icons-material/Edit';
+import axios from 'axios';
+import { formatDate } from './utils';
+
+const BASE_API_URL = 'http://localhost:3333/api';
+
+export function TodoItem(props) {
+  const {
+    id,
+    title: initialTitle,
+    author,
+    createdAt,
+    note: initialNote,
+  } = props.todo;
+  const [isDone, setIsDone] = useState(props.todo.isDone || false);
+  const [doneDate, setDoneDate] = useState(props.todo.doneDate || null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [title, setTitle] = useState(initialTitle);
+  const [note, setNote] = useState(initialNote);
+
+  function handleRemoveClick() {
+    axios
+      .delete(BASE_API_URL + '/todo/' + id)
+      .then((response) => {
+        console.log('Usunięto zadanie');
+        props.getTodoList();
+      })
+      .catch((error) => {
+        console.error('Błąd podczas usuwania zadania:', error);
+      });
+  }
+
+  function handleDoneClick() {
+    if (!isDone) {
+      setIsDone(true);
+      setDoneDate(new Date().toISOString());
+
+      axios
+        .put(`${BASE_API_URL}/todo/${id}/markAsDone`)
+        .then((response) => {
+          console.log('Zadanie zostało oznaczone jako wykonane na serwerze');
```

Comment:
Będziesz jeszcze kończył sobie to zadanie Todo List? Tu widzę jeszcze nie wszystko obsłużone.
Tak ogólnie kod wygląda bardzo dobrze, good job 👍